test/manifold/deferred_test.clj

```diff
@@ -213,7 +213,11 @@
   ;; body produces a delayed error result
   (let [ex (Exception.)]
     (is (= ex @(capture-error
-                 (d/loop [] (future-error ex)))))))
+                 (d/loop [] (future-error ex))))))
+
+  ;; destructuring works for loop parameters
+  (is @(capture-success
+        (d/loop [[a & b] [true false]] a))))
```

Comment:
Would it be good to add some more tests of the destructuring here, doing `recur`, taking the value on the `&`, e.t.c.?